core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraFromRjsBuilder.kt

```diff
@@ -84,31 +86,49 @@ class RawInfraFromRjsBuilderImpl : RawInfraBuilder {
         return sectionDistanceSortedChunkMap
     }
 
+    // TODO remove this accessor once useless in adapter
+    fun getTrackNodePool(): StaticPool<TrackNode, TrackNodeDescriptor> {
+        return trackNodePool
+    }
+
+    private fun getSectionIdx(sectionName: String): TrackSectionId {
+        return sectionNameToIdxMap[sectionName]
+            ?: throw OSRDError.newInfraLoadingError(
+                ErrorType.InfraHardLoadingError,
+                "Accessing track-section from unregistered name $sectionName"
+            )
+    }
+
     private fun getSectionDistanceSortedChunks(
         sectionName: String
     ): TreeMap<Distance, TrackChunkId> {
-        val sectionIdx =
-            sectionNameToIdxMap[sectionName]
-                ?: throw OSRDError.newInfraLoadingError(
-                    ErrorType.InfraHardLoadingError,
-                    "Accessing track-section from unregistered name $sectionName"
-                )
+        val sectionIdx = getSectionIdx(sectionName)
         return sectionDistanceSortedChunkMap[sectionIdx]
             ?: throw OSRDError.newInfraLoadingError(
                 ErrorType.InfraHardLoadingError,
                 "Accessing sorted chunks from unregistered track-section idx $sectionIdx (name: $sectionName)"
             )
     }
 
-    override fun movableElement(
+    fun getSectionEndpointIdx(sectionName: String, endpoint: Endpoint): EndpointTrackSectionId {
+        return EndpointTrackSectionId(getSectionIdx(sectionName), endpoint)
+    }
+
+    fun node(
         name: String,
         delay: Duration,
-        init: MovableElementDescriptorBuilder.() -> Unit
+        ports: StaticPool<TrackNodePort, EndpointTrackSectionId>,
+        configs: StaticPool<TrackNodeConfig, TrackNodeConfigDescriptor>
     ): TrackNodeId {
-        val movableElementBuilder = MovableElementDescriptorBuilderImpl(name, delay)
-        movableElementBuilder.init()
-        val movableElement = movableElementBuilder.build()
-        return trackNodePool.add(movableElement)
+        val nodeIdx = trackNodePool.add(TrackNodeDescriptor(name, delay, ports, configs))
+        for (portIdx in ports) {
+            val port = ports[portIdx]
+            assert(nodeAtEndpoint.get(port) == null) {
+                "Assertion failed: section ${trackSectionPool[port.value].name}.${port.endpoint} is referenced multiple times by a node port"
+            }
+            nodeAtEndpoint.getOrPut(port) { nodeIdx }
```

Comment:
```suggestion
            assert(nodeAtEndpoint[port] == null) {
                "Assertion failed: section ${trackSectionPool[port.value].name}.${port.endpoint} is referenced multiple times by a node port"
            }
            nodeAtEndpoint[port] = nodeIdx
```

`getOrPut` is a little misleading on the intent here

Reply:
Done in https://github.com/osrd-project/osrd/commit/c4d0a6a9f7a45fa3697f8a392413181cd6cc6e5d